include/sys/dmu_objset.h

```diff
@@ -197,6 +197,9 @@ struct objset {
 	dmu_objset_upgrade_cb_t os_upgrade_cb;
 	boolean_t os_upgrade_exit;
 	int os_upgrade_status;
+
+	/* prefetch */
+	boolean_t os_prefetch;
```

Comment:
You are adding field to be accessed for each block I/O request.  I wonder whether it will be a permanent cache miss or whether it can be aggregated with something already accessed, like os_spa or something else useful and read-only.

Reply:
Good question. It is the first time I add a property, and I just modeled it after `secondarycache`, which uses the `os_secondary_cache` field. I don't particularly like to add a new field, but I think the rationale is that, being a per-I/O request, looking up prefetch should be as fast as possible. Any suggestions?

Reply:
Done. I also moved `prefetch_table[]` just after `cache_table[]` in `zfs_prop.c`